rest_docs/rest-catalog-open-api-v0.1.yaml

```diff
@@ -0,0 +1,761 @@
+          schema:
+            type: boolean
+            default: false
+      responses:
+        '200':
+          description: OK
+          content:
+            'application/json':
+              schema:
+                type: boolean
+    head:
+      tags:
+        - Catalog API
+      summary: Check if a table with a given identifier exists
+      operationId: tableExists
+      description:
+        Check if a table exists within a given namespace. Returns the standard response with `true` when found. Will return a TableNotFound error if not present. Can change to returning a 200 with a body of `false` if not found, but that does add more wok on the client.
+      parameters:
+        - name: namespace
+          in: path
+          required: true
+          schema:
+            type: string
+        - name: table
+          in: path
+          required: true
+          schema:
+            type: string
+      responses:
+        '200':
+          description: OK
+        '412':
+          description: Table Not Found
+  /v1/tables:
+    get:
+      tags:
+        - Catalog API
+      summary: List all table identifiers underneath a given namespace
+      description: Return all table identifiers under this namespace
+      operationId: listTables
+      parameters:
+        - name: namespace
+          description: Namespace under which to get identifiers. Can be one or more levels.
+          in: query
+          required: false
+          schema:
+            type: string
+          examples:
+            singlepart_namespace:
+              value: "prod"
+            # Note that this one would need to be normalized due to the dots. Maybe the route should change.
+            multipart_namespace:
+              value: "prod.accounting"
+        # TODO - There's a much more native way to handle pagination
+        - name: limit
+          description: number of values to return in one request
+          in: query
+          required: false
+          schema:
+            type: integer
+          example: 10
+        - name: offset
+          description: Place in the response to continue from if paginating
+          in: query
+          schema:
+            type: integer
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ListTablesResponse'
+    post:
+      tags:
+        - Catalog API
+      summary: Create a table with the identifier given in the body
+      operationId: createTable
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/CreateTableRequest'
+        required: true
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/CreateTableResponse'
+  /v1/tables/renameTable:
```

Comment:
I think we don't need a explicit `renameTable` path, rename table can be expressed as a part of PUT `/tables/{table}` where the update payload is a different table name.

Reply:
I would be ok with that. I believe I was trying to balance not overloading the same route with too many verbs, but that's not necessarily an issue.

Let me see what I can do. 🙂

Reply:
Per @rdblue's request, I moved it back from `PUT /tables/{table}` to an explicit `POST /v1/tables/renameTable`.

This way, both table identifiers are in the request body and there isn't any ambiguity about which table in the route corresponds to the source name or target name, etc.